Destiny.ScrimTracker.Api/Controllers/MatchController.cs

```diff
@@ -0,0 +1,131 @@
+            ViewData.Add("PlayersPerTeam", playersPerTeam);
+
+            var match = new CreateMatchFormModel();
+
+            var teams = new List<Team>();
+            for (var i = 0; i < numOfTeams; i++)
+            {
+                teams.Add(new Team());
+                teams[i].MatchResults = new List<MatchResult>();
+                for (var j = 0; j < playersPerTeam; j++)
+                {
+                    teams[i].MatchResults.ToList().Add(new MatchResult());
+                }
+            }
+
+            match.Teams = teams;
+            return View(match);
+        }
+
+        [HttpPost]
+        [Authorize]
+        [Route("add_match")]
+        public IActionResult FormatRequestAndAddMatch([FromQuery] int numOfTeams, [FromQuery] int playersPerTeam)
+        {
+            var createMatchRequest = new CreateMatchRequest();
+
+            createMatchRequest.MatchType = Enum.Parse<MatchType>(Request.Form["MatchType"]);
+            var teams = new List<MatchTeam>();
+            for (var i = 0; i < numOfTeams; i++)
+            {
+                var team = new MatchTeam();
+                team.Name = Request.Form[$"Team[{i}].Name"];
+                team.TeamScore = int.Parse(Request.Form[$"Team[{i}].TeamScore"]);
+                
+                var guardianResults = new List<GuardianMatchResult>();
+                for (var j = 0; j < playersPerTeam; j++)
+                {
+                    var results = new GuardianMatchResult
+                    {
+                        Kills = int.Parse(Request.Form[$"Team[{i}].MatchResults[{j}].Kills"]),
+                        Assists = int.Parse(Request.Form[$"Team[{i}].MatchResults[{j}].Assists"]),
+                        Deaths = int.Parse(Request.Form[$"Team[{i}].MatchResults[{j}].Deaths"]),
+                        GuardianName = Request.Form[$"Team[{i}].MatchResults[{j}].GuardianName"]
+                    };
+
+                    if (results.Deaths == 0)
+                    {
+                        results.Efficiency = results.Kills + results.Assists;
+                    }
+                    else
+                    {
+                        results.Efficiency = (double) (results.Kills + results.Assists) / results.Deaths;
+                    }
+                    Console.WriteLine($"Calculated Eff = {results.Efficiency:F4}");
+                    guardianResults.Add(results);
+                }
+
+                team.GuardianMatchResults = guardianResults;
+                teams.Add(team);
+            }
+
+            createMatchRequest.Teams = teams;
+
+            _matchService.CreateMatch(createMatchRequest.ToMatch(), createMatchRequest.Teams);
+            return RedirectToAction("Index");
+        }
+
+        [HttpPost]
+        [Authorize]
+        public string Post(CreateMatchRequest request)
+        {
+            var match = request.ToMatch();
+            return _matchService.CreateMatch(match, request.Teams);
+        }
```

Comment:
When is this method used vs. the method above?

Reply:
This is left over from when this controller was an API controller. Will remove

Reply:
Same as the comments above. Will take care of this in the other PR